src/player/Player.php

```diff
@@ -368,8 +369,11 @@ function() : void{
 			}
 		));
 
-		$this->firstPlayed = $nbt->getLong(self::TAG_FIRST_PLAYED, $now = (int) (microtime(true) * 1000));
-		$this->lastPlayed = $nbt->getLong(self::TAG_LAST_PLAYED, $now);
+		$createDateTimeImmutable = static function(string $tag) use ($nbt) : DateTimeImmutable{
+			return new DateTimeImmutable('@' . $nbt->getLong($tag, (int) (microtime(true) * 1000)) / 1000);
```

Comment:
This is now creating two different timestamps. Previously, `$now` was reused to ensure that the two timestamps would be identical for new players.